test/tests/functional/pbs_cgroups_hook.py

```diff
@@ -2080,6 +2193,352 @@ def test_cgroup_use_hierarchy(self):
+            self.skipTest('There are not enough free vnodes to run the test')
+
+        # instantiate queuejob hook
+        hook_event = 'queuejob'
+        hook_name = 'qjob'
+        a = {'event': hook_event, 'enabled': 'true'}
+        self.server.create_import_hook(hook_name, a, self.qjob_hook_body)
+
+        # instantiate execjob_launch hook
+        hook_event = 'execjob_launch'
+        hook_name = 'launch'
+        a = {'event': hook_event, 'enabled': 'true'}
+        self.keep_select = 'e.job.Resource_List["site"]'
+        self.server.create_import_hook(
+            hook_name, a, self.launch_hook_body % (self.keep_select))
+
+        # Submit a job that requires 2 nodes
+        j = Job(TEST_USER)
+        j.create_script(self.job_scr2)
+        jid = self.server.submit(j)
+
+        # Check the exec_vnode while in substate 41
+        self.server.expect(JOB, {ATTR_substate: '41'}, id=jid)
+        self.server.expect(JOB, 'exec_vnode', id=jid, op=SET)
+        job_stat = self.server.status(JOB, id=jid)
+        execvnode1 = job_stat[0]['exec_vnode']
+        self.logger.info("initial exec_vnode: %s" % execvnode1)
+        initial_vnodes = execvnode1.split('+')
+
+        # Check the exec_vnode after job is in substate 42
+        self.server.expect(JOB, {ATTR_substate: '42'}, offset=20, id=jid)
+        self.server.expect(JOB, 'exec_vnode', id=jid, op=SET)
+        job_stat = self.server.status(JOB, id=jid)
+        execvnode2 = job_stat[0]['exec_vnode']
+        self.logger.info("pruned exec_vnode: %s" % execvnode2)
+        pruned_vnodes = execvnode2.split('+')
+
+        # Check that the pruned exec_vnode has one less than initial value
+        self.assertEqual(len(pruned_vnodes) + 1, len(initial_vnodes))
+
+        # Find the released vnode
+        for vn in initial_vnodes:
+            if vn not in pruned_vnodes:
+                rel_vn = vn
+        vnodeB = rel_vn.split(':')[0].split('(')[1]
+        self.logger.info("released vnode: %s" % vnodeB)
+
+        # Submit a second job requesting the released vnode, job runs
+        j2 = Job(TEST_USER,
+                 {ATTR_l + '.select': '1:ncpus=1:mem=2gb:vnode=%s' % vnodeB})
+        jid2 = self.server.submit(j2)
+        self.server.expect(JOB, {ATTR_state: 'R'}, offset=20, id=jid2)
+
+    def test_cgroup_sismom_resize_fail(self):
+        """
+        Verify that exec_vnode values are trimmed
+        when execjob_launch hook prunes job via release_nodes(),
+        exec_job_resize failure in sister mom,
+        tolerate_node_failures=job_start
+        """
+        # Test requires 2 nodes
+        if len(self.moms) != 2:
+            self.skipTest('Test requires two Moms as input, '
+                          'use -p moms=<mom1:mom2>')
+        self.load_config(self.cfg7)
+
+        # Check that moms have the required free vnodes
+        if self.check_req_rjs() == 1:
+            self.skipTest('There are not enough free vnodes to run the test')
```

Comment:
Please add detail to this message as before.

Reply:
I have changed the skipTest comment to be more explicit - here and in two more tests.